owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/files/DownloadRemoteFileOperation.kt

```diff
@@ -131,7 +131,7 @@ class DownloadRemoteFileOperation(
                     }
                 }
 
-                if (transferred == totalToTransfer) {  // Check if the file is completed
+                if (totalToTransfer == -1L || transferred == totalToTransfer) {  // Check if the file is completed
```

Comment:
Hey @appiekap653!

No, we're not checking if the download finishes with all the bytes for these cases in which `Content-Length` is not received. That's why we show an indeterminate progress bar 😄. In any case, this is a hotfix for the problem, the elegant solution will come later. Our idea is not relying on `Content-Length` anymore and check the total size (so that we're able to check the progress) from the metadata of the file in some way, since `Content-Length` might not express correctly the size of the file (for example, if the response was compressed in some way).

Reply:
Does there excists no situation where `savedfile` is set `True` while it isn't saved at all, with this modification? and the rest of the code about the modification date is still executed then.

Reply:
Those cases are covered some lines before this. If there is a crash in the server, a cancellation will be requested and so an `OperationCancelledException` will be thrown (line 120), and if disk space is full, the `write` method will throw an exception (line 123)